substratest/assets.py

```diff
@@ -168,15 +139,54 @@ def load(cls, d):
             raise errors.TError(f"cannot parse asset `{d}`") from e
 
 
-class _Asset(_DataclassLoader, abc.ABC):
+class _InternalStruct(pydantic.BaseModel, _DataclassLoader, abc.ABC):
+    """Internal nested structure"""
+
+
+class _Asset(_InternalStruct, abc.ABC):
     """Represents assets stored in the Substra framework.
 
     Convert a dict with camel case fields to a Dataclass.
     """
 
 
-@dataclasses.dataclass(frozen=True)
```

Comment:
I see that frozen has disappeared, did you try replacing it with pydantic faux immutability? https://pydantic-docs.helpmanual.io/usage/models/#faux-immutability

Reply:
I just tried and it works, I'll add it! 🙂